### PR TITLE
Undef Windows UMD macros that conflict with iOS

### PR DESCRIPTION
**Pull request by mnithish**

### CODE REVIEW

**Comment by mnithish** [comment]
@memontic-ms is added to the review. #Closed

**Comment by mnithish** [comment]
@bbowman is added to the review. #Closed

**Comment by mnithish** [comment]
@DHowett-MSFT is added to the review. #Closed

File: include/WOCStdLib/winnt.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#pragma once
+
+#include_next <winnt.h>
+#ifdef __OBJC__
+#undef DELETE
```
Comment (memontic-ms) [inline include/WOCStdLib/winnt.h]: Check #ifdef DELETE beforehand

**Comment by memontic-ms** [comment]
:shipit:

File: include/WOCStdLib/winnt.h (hunk 1 above)
Comment (mnithish) [inline include/WOCStdLib/winnt.h]: Good point. Done

---
In reply to: [74333399](https://github.com/Microsoft/WinObjC/pull/866#discussion_r74333399) [](ancestors = 74333399)

**Comment by DHowett-MSFT** [comment]
:shipit:

File: include/WOCStdLib/winnt.h (hunk 1 above)
Comment (bbowman) [inline include/WOCStdLib/winnt.h]: the "new hotness" for this is to use a pragma push which will push the old def onto to macro stack. You can then undef / redef things to your heart's content and pop at the end. This a convenient way to make sure the window's header gets what it wants but doesn't corrupt state down the line without the big hammer of killing anyone else's DELETE macro ... 

---
In reply to: [74333570](https://github.com/Microsoft/WinObjC/pull/866#discussion_r74333570) [](ancestors = 74333570,74333399)

**Comment by bbowman** [comment]
:clock1:

File: include/WOCStdLib/winnt.h (hunk 1 above)
Comment (mnithish) [inline include/WOCStdLib/winnt.h]: Good to know. Will change to that Brian

---
In reply to: [74335367](https://github.com/Microsoft/WinObjC/pull/866#discussion_r74335367) [](ancestors = 74335367,74333570,74333399)

**Pull request merged**